frontend/src/components/header/MenuOverlay.js

```diff
@@ -96,18 +97,23 @@ class MenuOverlay extends Component {
     if (targetValue) {
       this.setState({
         query: e.target.value,
+        pendingQuery: e.target.value ? true : false,
       });
 
       queryPathsRequest(e.target.value, 9)
         .then((response) => {
-          this.setState({
-            queriedResults: flattenLastElem(response.data),
-          });
+          if (e.target.value === this.queryString) {
+            this.setState({
+              queriedResults: flattenLastElem(response.data),
+              pendingQuery: false,
+            });
+          }
         })
         .catch((err) => {
           if (err.response && err.response.status === 404) {
             this.setState({
               queriedResults: [],
+              pendingQuery: false,
```

Comment:
`query:''`?

Reply:
compare the query with the one from the state and set it only if they are equal

Reply:
query already set on l:99 so no need to set it again in here in case of error

Reply:
ohh boy, let me explain it to you by voice